txscript: add test case for ExtractPkScriptAddrs()

JFixby opened the pull request:
<s>Ensure MainNetParams.OrganizationPkScript parameter matches the organization address: Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx 

Ensure SimNetParams.OrganizationPkScript parameter matches the organization address: ScuQxvveKGfpG1ypt6u27F99Anf7EW3cqhq 

Prepare the same test for the test-net.
</s>

Test ExtractPkScriptAddrs() on Decred main net organizational script and developer fund address values.

Simply checks
`Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx = ExtractPkScriptAddrs(hexDecode("a914f5916158e3e2c4551c1796708db8367207ed13bb87"))`

davecgh commented:
This makes no sense at all to me.  It's testing a hard-coded value is a hard-coded value.

davecgh commented:
As discussed in slack, it is reasonable to test if the value properly decodes, but it shouldn't be tested against hard-coded values.

JFixby commented:
Basically this test says:

"Yes, this address `Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx` 
the one you see in the blockchain explorer as the receviver of the def funs
this address is in fact equal to the `ExtractPkScriptAddrs(hexDecode("a914f5916158e3e2c4551c1796708db8367207ed13bb87"))`
where the `hexDecode("a914f5916158e3e2c4551c1796708db8367207ed13bb87")`
is the `OrganizationPkScript` value in the network parametes."

"Yes, to start a new network you need
to set the `OrganizationPkScript` parameter,
and I will help you to check desired org address of your network,
don't forget to set it here, but I will remind you anyway."

davecgh commented:
Address decoding is already tested where addresses are defined in `dcrutil`.   Namely, in `address_test.go`.  If you really want to add this specific address and script combination, that would be the place to do it.

JFixby commented:
I need a little help here.

1) I searched for `ExtractPkScriptAddrs()` tests, and `txscript/standard_test.go` is the only one testing the function. So I placed it there. The only issue was to transform the address string to `dcrutil.Address`. Decoding function returns 2 arguments, so I added the helper to wrap the call and to handle the error.

2) I checked the `dcrutil/address_test.go` as you pointed. It seems a little bit confusing to me. So the goal of this PR is to establish relationship between the `a914f5916158e3e2c4551c1796708db8367207ed13bb87` and `Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx`.
I couldn't find how to properly integrate these constants in the test case.
I added the following case:
```go
{
			name:    "MainNet organizational script and address",
			addr:    "Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx",
			encoded: "Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx",
			valid:   true,
			result: dcrutil.TstAddressPubKeyHash(
				[ripemd160.Size]byte{
					0x27, 0x89, 0xd5, 0x8c, 0xfa, 0x09, 0x57, 0xd2, 0x06, 0xf0,
					0x25, 0xc2, 0xaf, 0x05, 0x6f, 0xc8, 0xa7, 0x7c, 0xeb, 0xb0},

				chaincfg.MainNetParams.PubKeyHashAddrID),
			f: func() (dcrutil.Address, error) {
				pkHash := []byte{
					0x27, 0x89, 0xd5, 0x8c, 0xfa, 0x09, 0x57, 0xd2, 0x06, 0xf0,
					0x25, 0xc2, 0xaf, 0x05, 0x6f, 0xc8, 0xa7, 0x7c, 0xeb, 0xb0}
				return dcrutil.NewAddressPubKeyHash(pkHash,
					&chaincfg.MainNetParams, dcrec.STEcdsaSecp256k1)
			},
			net: &chaincfg.MainNetParams,
		},
```

Unexpectedly to me it passes:
 - there is no occurrence of the pkScript in it
 - it passes even the `result` and `f` are completely irrelevant bytes from another test. Probably this means that test doesn't actually establish relationship between `addr` and `encoded` on one side and the  `result` and `f`, seems like they are irrelevant to each other, and this looks suspicious to me. Probably there is something wrong with the test, or I'm just not getting it.

JFixby commented:
I'll close it until I figure this out.
